Enhance developer docs on primitive validation and developer documents for debugging AaC via Gitpod (#557)
DevOps-MBSE/AaC: docs/docs/dev_guide/aac_validation.md

Reviewer: Any thoughts on this?

Author: Oh, yeah. Captured it here in #593

        - behavior
```

### Definition Validator Plugins
Each plugin can register multiple definition validators, each of which provides a validation function that's used to validate any corresponding definition structures. It's recommended that plugins have a small, narrow scope of responsibility by focusing on a single validation, but multiple validators can be registered. Definition validators are registered via the plugin function `plugin.register_definition_validations(...)`

> Each validator plugin should provide one validation function with the following signature:

Author: Ah, I think the doc doesn't make sense. The validation function is the validator's interface.

```suggestion
Each schema validator must implement a validation function with the following signature/interface:
```
